common/tier4_autoware_utils/include/tier4_autoware_utils/ros/published_time_publisher.hpp

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TIER4_AUTOWARE_UTILS__ROS__PUBLISHED_TIME_PUBLISHER_HPP_
+#define TIER4_AUTOWARE_UTILS__ROS__PUBLISHED_TIME_PUBLISHER_HPP_
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <autoware_common_msgs/msg/published_time.hpp>
+
+namespace tier4_autoware_utils
+{
+using autoware_common_msgs::msg::PublishedTime;
+
+class PublishedTimePublisher
+{
+public:
+  explicit PublishedTimePublisher(
+    rclcpp::Node * node, const std::string & name = "~/debug/published_time",
+    const rclcpp::QoS & qos = rclcpp::QoS(1))
+  {
+    pub_published_time_ = node->create_publisher<PublishedTime>(name, qos);
+  }
+
+  void publish(const rclcpp::Time & header_stamp) const
+  {
+    PublishedTime published_time;
```

Comment:
We can add condition check for subscriber count at the beginning of this function and do nothing if there are no subscribers.